mne_sandbox/connectivity/simulation.py

```diff
@@ -0,0 +1,75 @@
+import numpy as np
+
+
+def simulate_pac_signal(time, freq_phase, freq_amp, max_amp_lo=2.,
+                        max_amp_hi=.5, frac_pac=.1, amp_noise_lo=None,
+                        amp_noise_hi=None, mask_pac_times=None):
+    """Simulate a signal with phase-amplitude coupling according to [1].
+
+    Parameters
+    ----------
+    time : array, shape (n_times,)
+        The times for the signal (which implicitly defines the sampling
+        frequency).
+    freq_phase : float
+        The frequency of the low-frequency phase that modulates amplitude.
+    freq_amp : float
+        The frequency of the high-frequency amplitude that is modulated by
+        phase.
+    max_amp_lo : float
+        The maximum amplitude for the low-frequency phase signal
+    max_amp_hi : float
+        The maximum amplitude for the high-frequency amplitude signal
+    frac_pac : float, between (0, 1)
+        The fraction of the high-frequency amplitude that is modulated by
+        low-frequency phase
+    amp_noise_lo : float | None
+        The amplitude of noise added to the low-frequency signal.
+    amp_noise_hi : float | None
+        The amplitude of noise added to the high-frequency signal.
+    mask_pac_times : array, dtype bool, shape (n_times,) | None
+        Whether to mask specific times to induce PAC. Values where
+        `mask_pac_times` is False will have `frac_pac` set to 0.
+        If None, all times have PAC.
+
+    Returns
+    -------
+    signal : array, shape (n_times,)
+        The simulated PAC signal w/ both low and high frequency components
+    phase_signal : array, shape (n_times,)
+        The low-frequency phase
+    amp_signal : array, shape (n_times,)
+        The high-frequency amplitude
+
+    References
+    ----------
+    .. [1] Tort, et al. "Measuring Phase-Amplitude Coupling Between Neuronal
+           Oscillations of Different Frequencies." Journal of Neurophysiology,
+           vol. 104, issue 2, 2010.
+    """
+    if mask_pac_times is None:
+        mask_pac_times = np.ones_like(time).astype(bool)
+    amp_noise_lo = max_amp_lo / 4. if amp_noise_lo is None else amp_noise_lo
+    amp_noise_hi = max_amp_hi / 4. if amp_noise_hi is None else amp_noise_hi
```

Comment:
good idea, will do

On Fri, May 13, 2016 at 4:26 PM, Jean-Rémi KING <notifications@github.com>
wrote:

> In mne_sandbox/connectivity/simulation.py
> <https://github.com/mne-tools/mne-sandbox/pull/7#discussion_r63263893>:
>
> > +        The simulated PAC signal w/ both low and high frequency components
> > +    phase_signal : array, shape (n_times,)
> > +        The low-frequency phase
> > +    amp_signal : array, shape (n_times,)
> > +        The high-frequency amplitude
> > +
> > +    References
> > +    ----------
> > +    .. [1] Tort, et al. "Measuring Phase-Amplitude Coupling Between Neuronal
> > +           Oscillations of Different Frequencies." Journal of Neurophysiology,
> > +           vol. 104, issue 2, 2010.
> > +    """
> > +    if mask_pac_times is None:
> > +        mask_pac_times = np.ones_like(time).astype(bool)
> > +    amp_noise_lo = max_amp_lo / 4. if amp_noise_lo is None else amp_noise_lo
> > +    amp_noise_hi = max_amp_hi / 4. if amp_noise_hi is None else amp_noise_hi
>
> yes, I would add an snr parameter.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/mne-tools/mne-sandbox/pull/7/files/88eb3c34c76f70ef766893583e5cfd66124849df#r63263893>
>